### ISSUE TITLE
typing-only-third-party-import (TCH002) doesn't catch compound imports

### ISSUE THREAD

**Issue by korverdev** [astral-sh/ruff#13060]
When some of the imports from the same module are only used for type hinting, but not others, Ruff won't recommend any of the imports be moved to a type checking block when [TCH002](https://docs.astral.sh/ruff/rules/typing-only-third-party-import/) is enabled.

Minimum recreatable example:
```python
from test_module import a, b

c: b = a()
```

Expected recommendation:
```python
from test_module import a

if TYPE_CHECKING:
    from test_module import b
```

**Comment by AlexWaygood** [comment]
Hmm, why would you want this? To import a single member from `test_module`, Python needs to parse and execute the entire file `test_module.py` (assuming it's a single-file module). Splitting the import into two just slows down the startup time for your module (because Python now has to execute two import statements rather than one — imports are cached in `sys.modules`, but there's still a cost) and doubles the vertical space taken up by the imports.

Unless test_module.py has a fancy module-level `__getattr__` method that it uses to provide the symbols `a` and `b` lazily, I suppose... But that's something of a rare edge case?

**Comment by Daverball** [comment]
This should already be possible by turning on the strict setting, i.e. `lint.flake8-type-checking.strict = true`, the original flake8 plugin also allows you to set whether you want to flag this or not, but it's off by default, exactly for the reason that @AlexWaygood mentioned. There's not really any runtime benefit.

**Comment by korverdev** [comment]
I'm not really concerned about performance during type checking which is, if I'm not mistaken, the only time those performance concerns would apply. I'm more interested in the potential clarity that forcing all type check only imports into the same block gives me, but it sounds like there's a good reason for this not to be enabled by default. Thanks for pointing me at that option that I overlooked.

**Comment by AlexWaygood** [comment]
> I'm not really concerned about performance during type checking which is, if I'm not mistaken, the only time those performance concerns would apply.

Right, splitting it into two import statements likely won't slow things down at runtime if one of the import cycles is in an `if TYPE_CHECKING` block -- I misspoke. (However, putting some of the imports into a separate `if TYPE_CHECKING` branch also won't speed things up at runtime or break any runtime import cycles either -- and those are _generally_ the motivations for moving imports into `if TYPE_CHECKING` blocks.)

**Comment by korverdev** [comment]
@AlexWaygood A somewhat related question for you since you seem to be pretty active on this project. This wasn't documented as an option for that check in the documentation, which is why I overlooked it originally. Is there a reason for this you know of? Just wanted to check before I submit a PR to flesh out the documentation.

**Comment by AlexWaygood** [comment]
> @AlexWaygood A somewhat related question for you since you seem to be pretty active on this project. This wasn't documented as an option for that check in the documentation, which is why I overlooked it originally. Is there a reason for this you know of? Just wanted to check before I submit a PR to flesh out the documentation.

Not that I know of. We seem to already explicitly list nearly all the other `flake8-type-checking` options at https://docs.astral.sh/ruff/rules/typing-only-third-party-import/#options -- I think it makes sense to list `flake8-type-checking.strict` as well! [`exempt-modules`](https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_exempt-modules) could also possibly be listed.

A docs PR would be welcome, thanks!
